app/client/keybase/keystore.go

```diff
@@ -84,15 +83,19 @@ func (keybase *badgerKeybase) Create(passphrase, hint string) error {
 
 		return tx.Set(addrKey, keypairBz)
 	})
+	if err != nil {
+		return nil, err
+	}
 
-	return err
+	return keyPair, nil
 }
 
 // Create a new KeyPair from the private key hex string and store the serialised KeyPair encoding in the DB
 // Using the PublicKey.Address() return value as the key for storage
-func (keybase *badgerKeybase) ImportFromString(privKeyHex, passphrase, hint string) error {
-	err := keybase.db.Update(func(tx *badger.Txn) error {
-		keyPair, err := crypto.CreateNewKeyFromString(privKeyHex, passphrase, hint)
+func (keybase *badgerKeybase) ImportFromString(privKeyHex, passphrase, hint string) (crypto.KeyPair, error) {
```

Comment:
Optional NIT: This is a good opportunity to use named params (to avoid declaring keyPair as a param) but don't feel strongly about it.